### DISCUSSION TITLE
Auto add items to repeater with relationship()

### DISCUSSION THREAD

**Discussion by naseef** [filamentphp/filament#10178] [Help]
have a Select and a Repeater. The repeater is of a `relationship()`  I want to search for a product using the Select via afterStateUpdated and automatically add the the related models of the product to the Repeater. I am using the default create page of a resource. How can I do this?

**Comment by leandrocfe** [comment]
If a parent record hasn't been created yet, the corresponding relationship should not be accessible. However, if you wish to manually save the relationship, you can use the `dehydrated(true)` option within the Repeater field.

**Reply by naseef** [reply]
I just want to add repeater items ( and the form fields in repeater) based from the  `afterStateUpdated ` . I can add this if the repeater has no `relationship()` with the answer  [here](https://github.com/filamentphp/filament/discussions/9939#discussioncomment-7702407). I want the same functionality with the repeater having `relationship()`
